crates/turbopack-node/src/render/render_proxy.rs

```diff
@@ -44,105 +52,55 @@ pub async fn render_proxy(
         output_root,
         project_dir,
         /* debug */ false,
-    )
-    .await?;
-
-    let mut operation = match pool.operation().await {
-        Ok(operation) => operation,
-        Err(err) => {
-            return proxy_error(path, err, None).await;
-        }
-    };
+    );
 
-    match run_proxy_operation(
-        &mut operation,
+    let render = render_stream(
+        pool,
         data,
         body,
         intermediate_asset,
         intermediate_output_path,
         project_dir,
+        path,
     )
-    .await
-    {
-        Ok(proxy_result) => Ok(proxy_result.cell()),
-        Err(err) => Ok(proxy_error(path, err, Some(operation)).await?),
-    }
-}
-
-async fn run_proxy_operation(
-    operation: &mut NodeJsOperation,
-    data: RenderDataVc,
-    body: BodyVc,
-    intermediate_asset: AssetVc,
-    intermediate_output_path: FileSystemPathVc,
-    project_dir: FileSystemPathVc,
-) -> Result<ProxyResult> {
-    let data = data.await?;
-    // First, send the render data.
-    operation
-        .send(RenderProxyOutgoingMessage::Headers { data: &data })
-        .await?;
-
-    let mut body = body.await?.read();
-    // Then, send the binary body in chunks.
-    while let Some(data) = body.next().await {
-        operation
-            .send(RenderProxyOutgoingMessage::BodyChunk { data: &data? })
-            .await?;
-    }
-
-    operation.send(RenderProxyOutgoingMessage::BodyEnd).await?;
+    .await?;
 
-    let (status, headers) = match operation.recv().await? {
-        RenderProxyIncomingMessage::Headers {
-            data: ResponseHeaders { status, headers },
-        } => (status, headers),
-        RenderProxyIncomingMessage::Error(error) => {
-            bail!(
-                trace_stack(
-                    error,
-                    intermediate_asset,
-                    intermediate_output_path,
-                    project_dir
-                )
-                .await?
-            )
-        }
+    let mut stream = render.read();
+    let first = match stream.next().await {
```

Comment:
```suggestion
    let first = match stream.try_next().await? {
```

To propagate potential errors

Reply:
Done.